_posts/2022-08-14-data-pipelines-production-ready-great-expectations.md

```diff
@@ -0,0 +1,102 @@
+quality. Faulty data comes in, which can either cause data pipelines to fail, or
+possibly go unnoticed and cause inaccurate downstream reporting. Tacit knowledge
+is scattered among domain experts, technical experts, and the code and data
+itself. Maintenance becomes time-consuming and expensive. Documentation gets out
+of date and unreliable. This is why using data quality checks using libraries 
+like [Great Expectations](https://greatexpectations.io/) is so important when
+writing ETL applications.
+
+## Improve data quality in Trino with Great Expectations
+
+As data quality becomes more in the forefront of the Trino community, the Great
+Expectations and Trino communities have partnered to do some events together:
+* [Trino meetup to discuss Great Expectations](https://www.youtube.com/watch?v=pcqAOq3O3Ts&list=PLFnr63che7wZij92ynF_egatbsrH7by7T&index=3)
+* [Great Expectations meetup to discuss Trino](https://www.youtube.com/watch?v=4SieRmibb0U).
+* [Superconductive](https://superconductive.ai/) joined this year’s mini Trino 
+Summit event 
+[Cinco de Trino](https://www.youtube.com/watch?v=kfJ63DNbAuI&list=PLFnr63che7wYDHjUsmp43THLmAlqPDHlM)
+to showcase using 
+[managed solutions for Great Expectations and Trino](https://www.youtube.com/watch?v=9HE6LawCHP8&list=PLFnr63che7wYDHjUsmp43THLmAlqPDHlM&index=7).
+
+Today, we’re showcasing a demo that showcases a scenario with Trino running as
+the datalake query engine with multiple phases of data transformations on some
+Pokemon data sets. At each phase, we need to validate the data is in the correct
+schema, counts, and various other factors to validate.
+
+So in the context of this demo, we’re going to be using Trino as the query
+engine of a data lakehouse. For this, you can use various table formats. To
+index and read the data, we’re going to use the Hive table format for landing
+tables and Iceberg table format for the structure and consume tables. This is
+mainly because Trino currently doesn’t support reading CSV files from Iceberg 
+yet, but also conveys the level of flexibility you have in a data lake.
+
+## Trino and Great Expectations demo
+
+In this scenario, we’re going to ingest Pokemon pokedex data and Pokemon Go 
+spawn location data which lands as raw CSV files in our data lake. We then use
+Trino’s Hive catalog to read the data from the landing files and clean up and 
+optimize that raw data into more performant ORC files in the structure tables. 
+
+![](/assets/blog/data-pipelines-production-ready-great-expectations/trino-ge-lakehouse.svg)
+
+The last step is to join and transform the spawn data and pokedex data into a
+single table that is cleaned and ready to be utilized by a data analyst, data
+scientist, or other data consumer. Every area of the pipeline where the data is
+transformed opens up a vulnerability as the state can go from good to bed as
+infrastructure fails or is updated as newer versions of the pipeline roll out.
```

Comment:
The `as newer versions of the pipeline roll out` make this sentence _very_ long. If we can afford to drop it or split this sentence up, that might be better.

Reply:
Fixed by breaking up the sentence